### PR TITLE
Addition of psi4fockci plugin

### PR DESCRIPTION
**Pull request by shannonhouck**
## Description
This PR adds the [psi4fockci plugin](https://github.com/shannonhouck/psi4fockci) as an endorsed plugin, allowing people to perform generalized spin-flip and IP/EA calculations. ([Related to Issue #1838 here](https://github.com/psi4/psi4/issues/1838).)

## Todos
Notable points (developer or user-interest) that this PR has or will accomplish.
- [x] Added psi4fockci to endorsed_plugins.py
- [x] Added documentation
- [x] Added tests

## Questions
- [x] The addition of the psi4fockci tests seem to be breaking test_psi4_cas; it fails with a "Fatal Error: CIWavefunction: Iterations did not converge!" error. I presume this is because I'm somehow not cleaning variables correctly; I'm calling core.clean(), core.clean_options(), and core.clean_variables(). Is there one I'm missing?

## Checklist
- [x] Tests added for psi4fockci
- [x] [All or relevant fraction of full tests run](http://psicode.org/psi4manual/master/build_planning.html#how-to-run-a-subset-of-tests)

## Status
- [x] Ready for review
- [ ] Ready for merge

### CODE REVIEW

**Comment by loriab** [comment]
Thanks for working on this!

The test fail isn't your fault. DETCI [isn't reset-able](https://github.com/psi4/psi4/pull/1850/files#diff-ac285870f28f146a56a93bd3592ff776R217), and that has looming consequences for pytest testing and QCA use. Does your test run ok in random order and w/o the cas test `pytest ... test_psi4.py -n2 -k "not cas"`? You've correctly found all the clean statements, but it isn't necessary to include them explicitly b/c they're run before every test https://github.com/psi4/psi4/blob/master/tests/pytests/conftest.py#L9-L14

**Comment by shannonhouck** [comment]
Yes, it seems to run fine! I am calling DETCI multiple times within the test itself, since I wanted to test SF, SF-IP, and SF-EA cases; I can reduce it to just SF-IP if that will be more stable in the long run.

I've also just added some docs; let me know if there's anything else I need to do. Thanks!

File: doc/sphinxman/source/plugin_psi4fockci.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Installation
+~~~~~~~~~~~~
+
+* Download the plugin from the GitHub repository:
+
+  .. code-block:: bash
+
+     >>> git clone https://github.com/shannonhouck/psi4fockci.git
+
+* Once dowloaded, the plugin can be installed as follows:
```
[the remark this answers is not in the document]
Comment (shannonhouck) [reply doc/sphinxman/source/plugin_psi4fockci.rst]: Thanks, good catch!

File: tests/pytests/test_addons.py [hunk 2]
```diff
@@ -40,7 +40,8 @@
                                 reason="Psi4 not detecting plugin snsmp2. Build plugin if necessary and add to envvar PYTHONPATH (or rebuild Psi with -DENABLE_snsmp2)")
 using_resp = pytest.mark.skipif(psi4.addons("resp") is False,
                                 reason="Psi4 not detecting plugin resp. Build plugin if necessary and add to envvar PYTHONPATH (or rebuild Psi with -DENABLE_resp)")
-
```
[the remark this answers is not in the document]
Comment (shannonhouck) [reply tests/pytests/test_addons.py:43]: Do you mean line 45, or the line break within the `skipif()`? Either way, I'm happy to take the newline out.

**Comment by dgasmith** [comment]
@shannonhouck Gentle poke here, let us know if this will be make the Psi4 1.4 release or needs to be pushed.

File: tests/pytests/test_addons.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shannonhouck) [reply tests/pytests/test_addons.py:43]: Got it, thanks! Should be fixed now

File: tests/pytests/test_addons.py [hunk 3]
```diff
@@ -1030,6 +1031,34 @@ def test_resp():
...
+        add_opts=options )
+    assert psi4.compare_values(-108.250639579451, psi4.core.variable("CI ROOT 0 TOTAL ENERGY"), 7, "2SF-IP Energy")
+
+    wfn = psi4.energy('psi4fockci', new_charge=-1, new_multiplicity=2,
+        add_opts=options )
+    assert psi4.compare_values(-108.600832070267, psi4.core.variable("CI ROOT 0 TOTAL ENERGY"), 7, "2SF-EA Energy")
+
+    psi4.core.clean()
+    psi4.core.clean_options()
+    psi4.core.clean_variables()
```
[the remark this answers is not in the document]
Comment (shannonhouck) [reply tests/pytests/test_addons.py]: Good catch, thank you! Should be fixed now.

**Comment by shannonhouck** [comment]
Oops, thanks for the reminder! I'd love to add the things that @loriab suggested so Psi4FockCI can be built with Psi4. It should be a relatively quick fix, so I can try and get that done before tomorrow if that's okay for the 1.4 release timeline. After that's complete, I think everything should be good to go, assuming there aren't any additional issues.

**Comment by loriab** [comment]
I've got a `fockci` conda package built, so you can try adding it to https://github.com/psi4/psi4/blob/master/.azure-pipelines/azure-pipelines-linux.yml#L90 and seeing if your pytest runs in continuous integration.

**Comment by shannonhouck** [comment]
@loriab I've added the internal build like you suggested (drawing from the previous add-on installation files). It seems to compile without issues on my machine, but let me know if anything looks awry!

File: external/downstream/psi4fockci/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Format borrowed from sns-mp2 CMakeLists
+
+if(${ENABLE_psi4fockci})
+    if(NOT (${CMAKE_DISABLE_FIND_PACKAGE_psi4fockci}))
+        include(FindPythonModule)
+        find_python_module(psi4fockci ATLEAST 1.0.0 QUIET)
```
Comment (loriab) [inline external/downstream/psi4fockci/CMakeLists.txt:6]: Need to adjust version number here or it'll never be found.

Also, need to add `psi4fockci` to https://github.com/psi4/psi4/blob/master/external/downstream/CMakeLists.txt#L5 b/c as-is this file never executed.

Good news is that the conda pkg looks to be found and the test passes.

Comment (shannonhouck) [reply external/downstream/psi4fockci/CMakeLists.txt]: Good catch, thanks-- the version number should be updated now.

I think I added `psi4fockci` to the downstream CMakeLists.txt in a recent commit (https://github.com/psi4/psi4/pull/1855/commits/3881cb475e50a567a74d956d9c13ed56445729a5) but let me know if I've added it in the wrong place or need to change anything else.

Thanks so much for getting the conda package set up!!

Comment (loriab) [reply external/downstream/psi4fockci/CMakeLists.txt]: I see the downstream/CM change now -- maybe I hadn't refreshed before.

File: tests/pytests/test_addons.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (loriab) [reply tests/pytests/test_addons.py:43]: GH highlighting is a little weird here, but spacing looks resolved in her branch. https://github.com/shannonhouck/psi4/blob/master/tests/pytests/test_addons.py#L41-L46

**Pull request merged**